zipline/src/jsMain/kotlin/app/cash/zipline/ziplineFunctionJs.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2024 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+import app.cash.zipline.internal.bridge.ArgsListSerializer
+import app.cash.zipline.internal.bridge.OutboundService
+import app.cash.zipline.internal.bridge.ResultOrCallbackSerializer
+import app.cash.zipline.internal.bridge.ReturningZiplineFunction
+import app.cash.zipline.internal.bridge.SuspendCallback
+import app.cash.zipline.internal.bridge.SuspendingZiplineFunction
+import kotlinx.serialization.ContextualSerializer
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.AbstractEncoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.encodeToDynamic
+
+/**
+ * Returns a function that accepts and returns plain JavaScript objects, instead of regular Kotlin
+ * values. (The function's receiver will still receive regular Kotlin values as parameters, and
+ * return a regular Kotlin value as a result.)
+ *
+ * Use this in highly performance-sensitive code to reduce the amount of work required to call
+ * bridged functions on Zipline services.
+ */
+@ExperimentalSerializationApi
+fun <T : ZiplineService> ZiplineFunction<T>.asDynamicFunction(): (service: T, args: List<*>) -> Any? {
```

Comment:
Feels like this should take an array
```suggestion
fun <T : ZiplineService> ZiplineFunction<T>.asDynamicFunction(): (service: T, args: Array<*>) -> Any? {
```
which maps to a JS-native array. Figure if you're going for performance and all...

Reply:
Good call. Fixed!